src/mlpack/tests/main_tests/gmm_train_test.cpp

```diff
@@ -0,0 +1,377 @@
+  SetInputParam("trials", (int) 0); // Invalid.
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+// Checking that percentage is between 0 and 1.
+BOOST_AUTO_TEST_CASE(RefinedStartPercentageTest)
+{
+  arma::mat inputData(5, 10, arma::fill::randu);
+
+  SetInputParam("input", std::move(inputData));
+  SetInputParam("refined_start", true);
+
+  Log::Fatal.ignoreInput = true;
+  SetInputParam("percentage", (double) 2.0); // Invalid
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+
+  SetInputParam("percentage", (double) -1.0); // Invalid
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+
+  Log::Fatal.ignoreInput = false;
+}
+
+// Samplings must be positive.
+BOOST_AUTO_TEST_CASE(GmmTrainSamplings)
+{
+  arma::mat inputData(5, 10, arma::fill::randu);
+
+  SetInputParam("input", std::move(inputData));
+  SetInputParam("refined_start", true);
+  SetInputParam("samplings", (int) 0); // Invalid
+
+  Log::Fatal.ignoreInput = true;
+  BOOST_REQUIRE_THROW(mlpackMain(), std::runtime_error);
+  Log::Fatal.ignoreInput = false;
+}
+
+// Number of gaussians in the model trained from input model.
+BOOST_AUTO_TEST_CASE(GmmTrainNumberOfGaussian)
+{
+  arma::mat inputData(5, 10, arma::fill::randu);
+
+  SetInputParam("input", inputData);
+  SetInputParam("gaussians", (int) 2);
+
+  mlpackMain();
+
+  GMM* gmm = CLI::GetParam<GMM*>("output_model");
+
+  SetInputParam("input_model", gmm);
+
+  CLI::GetSingleton().Parameters()["input"].wasPassed = false;
+
+  SetInputParam("input", std::move(inputData));
+
+  mlpackMain();
+
+  GMM* gmm1 = CLI::GetParam<GMM*>("output_model");
+
+  BOOST_REQUIRE_EQUAL(gmm1->Gaussians(), (int) 2);
+}
+
+// Ensure that Noise affects the final result.
+BOOST_AUTO_TEST_CASE(GmmTrainNoisetest)
+{
+  arma::mat inputData(5, 10, arma::fill::randu);
+
+  SetInputParam("input", inputData);
+  SetInputParam("gaussians", (int) 2);
+  SetInputParam("noise", (double) 0.0);
+
+  mlpack::math::FixedRandomSeed();
+  mlpackMain();
+
+  GMM* gmm = CLI::GetParam<GMM*>("output_model");
+
+  ResetGmmTrainSetting();
+
+  SetInputParam("input", std::move(inputData));
+  SetInputParam("gaussians", (int) 2);
+  SetInputParam("noise", (double) 100.0);
+
+  mlpack::math::FixedRandomSeed();
+  mlpackMain();
+
+  GMM* gmm1 = CLI::GetParam<GMM*>("output_model");
+
+  arma::uvec sortedIndices = sort_index(gmm->Weights());
+
+  for (size_t k = 0; k < sortedIndices.n_elem; k++)
+    BOOST_REQUIRE(arma::norm(gmm->Component(sortedIndices[k]).Mean() -
+                      gmm1->Component(sortedIndices[k]).Mean()) > 1e-50 ||
+                  arma::norm(gmm->Component(sortedIndices[k]).Covariance() -
+                      gmm1->Component(sortedIndices[k]).Covariance()) > 1e-50);
+}
+
+// Ensure that Percentage affects the final result when refined_start is true.
+BOOST_AUTO_TEST_CASE(GmmTrainPercentageTest)
+{
+  arma::mat inputData(5, 10, arma::fill::randu);
```

Comment:
You might consider a larger dataset here; more samples can increase the chance that the two models will be different.  I think that this applies across all the tests that check for changes.